compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1431,10 +1431,8 @@ fn add_late_link_args(
         if let Some(args) = sess.target.late_link_args_dynamic.get(&flavor) {
             cmd.args(args);
         }
-    } else {
-        if let Some(args) = sess.target.late_link_args_static.get(&flavor) {
-            cmd.args(args);
-        }
+    } else if let Some(args) = sess.target.late_link_args_static.get(&flavor) {
+        cmd.args(args);
     }
```

Comment:
FWIW I feel like this change is hiding intent - previously the code was pretty clearly handling the dynamic and non-dynamic cases, but now it is less obvious IMO, at least for me.